plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -610,6 +614,30 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
                 .build();
     }
 
+    @Override
+    public void validateScan(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) handle;
+        Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
+        if (icebergTable.spec().isUnpartitioned()) {
+            return;
+        }
+        if (IcebergSessionProperties.isFilterOnPartitionTable(session)
+                && table.getEnforcedPredicate().equals(TupleDomain.all())) {
+            Set<IcebergColumnHandle> icebergConstraintColumns = table.getConstraintColumns().stream()
+                    .map(columnHandle -> (IcebergColumnHandle) columnHandle)
+                    .collect(Collectors.toSet());
+            List<String> par1 = icebergConstraintColumns.stream().map(IcebergColumnHandle::getName).toList();
+            List<String> par2 = icebergTable.spec().fields().stream().map(PartitionField::name).toList();
```

Comment:
Could this also give back fields which have void transform? (i.e. used to be a partition field, but not anymore). If so, we wouldn't expect users to apply predicates for those columns anymore

Reply:
A partition field with void transform will be selected (like `field_null`), in order not to bother users, we can filter out the `VoidTransform` partition.